src/request-pipeline/utils.ts

```diff
@@ -16,6 +16,25 @@ import { toReadableStream } from '../utils/buffer';
 import { PassThrough } from 'stream';
 import { getText, MESSAGE } from '../messages';
 import logger from '../utils/logger';
+import { getFormattedInvalidCharacters } from './http-header-parser';
+
+// An empty line that indicates the end of the header section
+// https://tools.ietf.org/html/rfc7230#section-3
+const HTTP_BODY_SEPARATOR = '\r\n\r\n';
+
+// Used to calculate the recommended maximum header size
+// See recommendMaxHeaderSize() below
+const HEADER_SIZE_MULTIPLIER            = 2;
+const HEADER_SIZE_CALCULATION_PRECISION = 2;
+
+// Calculates the HTTP header size in bytes that a customer should specify via the
+// --max-http-header-size Node option so that the proxy can process the site
+// https://nodejs.org/api/cli.html#cli_max_http_header_size_size
+// Example: 
+// (8211 * 2).toPrecision(2) -> 16 * 10^3 -> 16000
+function recommendMaxHeaderSize (currentHeaderSize: number): number {
+    return Number((currentHeaderSize * HEADER_SIZE_MULTIPLIER).toPrecision(HEADER_SIZE_CALCULATION_PRECISION));
```

Comment:
Can we remove the `Number` constructor call here?
```js
return (currentHeaderSize * HEADER_SIZE_MULTIPLIER).toPrecision(HEADER_SIZE_CALCULATION_PRECISION);
```

Reply:
[toPrecision](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/toPrecision) returns a string in exponential notation in this case

Reply:
Ok